File: C++/001. Two Sum.cpp
#include <iostream>
#include <vector>
#include <map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::map<int, int> m;
        std::vector<int> res (2, 0);
        for(int i = 0; i < int(nums.size()); i++) {
            if(m.find(nums[i]) != m.end()) {
                res[0] = m[nums[i]];
                res[1] = i;
                break;
            }
            m.insert(std::map<int, int>::value_type(target - nums[i], i));
        }
        return res;
    }
};

File: Go/001. Two Sum.go
package main

func twoSum(nums []int, target int) []int {
	N := len(nums)
	m := make(map[int]int)
	res := make([]int, 2, 2)
	for i := 0; i < N; i++ {
		if _, ok := m[nums[i]]; ok {
			res[0] = m[nums[i]]
			res[1] = i
			break
		}
		m[target - nums[i]] = i
	}
	return res
}